packages/react-core/src/components/Tabs/OverflowTab.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import styles from '@patternfly/react-styles/css/components/Tabs/tabs';
+import { css } from '@patternfly/react-styles';
+import AngleRightIcon from '@patternfly/react-icons/dist/esm/icons/angle-right-icon';
+import { Popper } from '../../helpers';
+import { Menu, MenuContent, MenuList, MenuItem } from '../Menu';
+import { TabsContext } from './TabsContext';
+import { TabProps } from './Tab';
+
+export interface OverflowTabProps extends React.HTMLProps<HTMLLIElement> {
+  /** Additional classes added to the overflow tab */
+  className?: string;
+  /** Element to append the menu displaying the overflow tabs to */
+  menuAppendTo?: HTMLElement | (() => HTMLElement);
+  /** The tabs that should be displayed in the menu */
+  overflowingTabs?: TabProps[];
+  /** Flag which shows the count of overflowing tabs when enabled */
+  showTabCount?: boolean;
+}
+
+export const OverflowTab: React.FunctionComponent<OverflowTabProps> = ({
+  className,
+  menuAppendTo,
+  overflowingTabs = [],
+  showTabCount,
+  ...props
+}: OverflowTabProps) => {
+  const menuRef = React.useRef<HTMLDivElement>();
+  const overflowTabRef = React.useRef<HTMLLIElement>();
+
+  const [isExpanded, setIsExpanded] = React.useState(false);
+
+  const { localActiveKey, handleTabClick } = React.useContext(TabsContext);
+
+  const handleMenuKeys = (ev: KeyboardEvent) => {
+    const menuContainsEventTarget = menuRef?.current?.contains(ev.target as Node);
+    if (isExpanded && menuContainsEventTarget && ev.key === 'Escape') {
+      setIsExpanded(!isExpanded);
+      overflowTabRef?.current.focus();
+    }
+  };
+
+  const handleClick = (ev: MouseEvent) => {
+    const clickIsOutsideMenu = !menuRef?.current?.contains(ev.target as Node);
+    const clickIsOutsideOverflowTab = !overflowTabRef?.current?.contains(ev.target as Node);
+    if (isExpanded && clickIsOutsideMenu && clickIsOutsideOverflowTab) {
+      setIsExpanded(false);
+    }
+  };
+
+  React.useEffect(() => {
+    window.addEventListener('click', handleClick);
+    window.addEventListener('keydown', handleMenuKeys);
+
+    return () => {
+      window.removeEventListener('click', handleClick);
+      window.removeEventListener('keydown', handleMenuKeys);
+    };
+  }, [isExpanded, menuRef, overflowTabRef]);
+
+  const selectedTab = overflowingTabs.find(tab => tab.eventKey === localActiveKey);
+  const tabTitle = selectedTab?.title ? selectedTab.title : 'More';
+
+  const toggleMenu = () => {
+    setIsExpanded(prevIsExpanded => !prevIsExpanded);
+    setTimeout(() => {
+      if (menuRef?.current) {
+        const firstElement = menuRef.current.querySelector('li > button,input:not(:disabled)');
+        firstElement && (firstElement as HTMLElement).focus();
+      }
+    }, 0);
+  };
+
+  const overflowTab = (
+    <li
```

Comment:
there is an a11y violation being thrown -`<li>` elements must be contained in a `<ul>` or `<ol>`
 it's possible you're just missing a `role='presentation'`here to fix this

Reply:
The rendered structure looks valid to me, but I noticed we have `role="presentation"` on the current tab and the overflow tab. When are those not children of a list parent?

<img width="1034" alt="Screen Shot 2022-07-27 at 4 32 45 PM" src="https://user-images.githubusercontent.com/35148959/181376442-5985ec6c-181c-4546-9e76-bd21ef8a178d.png">

Reply:
Since the wrapping `ul` element has a role='tablist', the `li`'s each need to have a presentation role and the buttons within them have the role='tab'

Reply:
@nicolethoen ah! Gotcha, I think I was looking at it wrong 😅